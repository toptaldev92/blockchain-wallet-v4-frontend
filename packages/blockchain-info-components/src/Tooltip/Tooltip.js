import React from 'react'
import styled from 'styled-components'
import { transparentize } from 'polished'
import { TextGroup } from '../Text'

const TooltipWrapper = styled.div`
  display: inline-flex;
  position: relative;
  width: 22px;
  margin: 2px;
`
const TooltipIcon = styled.div`
  display: inline-flex;
  justify-content: center;
  align-items: center;
  width: 16px;
  height: 16px;
  border-radius: 10px;
  background-color: ${props => props.theme[props.colors.backgroundColor]};
  color: ${props => props.theme[props.colors.foreColor]};
  border: 1px solid ${props => props.theme[props.colors.borderColor]};
  cursor: pointer;
  font-size: 12px;
  font-weight: 300;
`
const TooltipBox = styled(TextGroup)`
  position: absolute;
  bottom: 150%;
  left: -115px;
  width: 250px;
  display: ${props => props.displayed ? 'block' : 'none'};
  background-color: ${props => props.theme['gray-1']};
  color: ${props => props.theme['gray-5']};
  border: 1px solid ${props => props.theme['gray-2']};
  border-radius: 5px;
  padding: 10px;
  box-sizing: border-box;
  box-shadow: ${props => transparentize(0.65, props.theme['gray-6'])} 0px 3px 8px 0px;
  cursor: pointer;
  font-size: 13px;
  font-weight: 300;
  font-family: "GillSans", sans serif;

  &:before {
    content: '';
    display: block;
    position: absolute;
    left: 115px;
    top: 100%;
    width: 0;
    height: 0;
    border: 10px solid transparent;
    border-top-color: ${props => props.theme['gray-2']};
  }

  &:after {
    content: '';
    display: block;
    position: absolute;
    left: 116px;
    top: 100%;
    width: 0;
    height: 0;
    border: 9px solid transparent;
    border-top-color: ${props => props.theme['gray-1']};
  }
`

const selectColors = displayed => displayed
  ? { foreColor: 'white', backgroundColor: 'brand-primary', borderColor: 'brand-primary' }
  : { foreColor: 'gray-5', backgroundColor: 'white', borderColor: 'gray-2' }

class Tooltip extends React.Component {
  constructor (props) {
    super(props)
    this.state = { displayed: false }
    this.handleClick = this.handleClick.bind(this)
  }

  handleClick () {
    this.setState({ displayed: !this.state.displayed })
  }

  render () {
    const icon = this.state.displayed ? 'X' : '?'
    const colors = selectColors(this.state.displayed)

    return (
      <TooltipWrapper>
        <TooltipIcon displayed={this.state.displayed} colors={colors} onClick={this.handleClick}>{icon}</TooltipIcon>
        <TooltipBox displayed={this.state.displayed} onClick={this.handleClick}>{this.props.children}</TooltipBox>
      </TooltipWrapper>
    )
  }
}

export default Tooltip
