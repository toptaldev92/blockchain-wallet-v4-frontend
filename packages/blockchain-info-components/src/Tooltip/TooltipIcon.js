import styled from 'styled-components'

import { Icon } from '../Icons'

const TooltipIcon = styled(Icon)`
  display: inline-flex;
  margin-left: 5px;
`

export default TooltipIcon
