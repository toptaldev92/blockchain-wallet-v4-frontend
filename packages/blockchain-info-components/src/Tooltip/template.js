import React from 'react'
import styled from 'styled-components'

import ReactTooltip from 'react-tooltip'

const StyledTip = styled(ReactTooltip)`
  cursor: pointer;
  font-weight: 400;
  text-align: left;
  padding: 7px 12px;
  text-transform: none;
  z-index: 2000 !important;
  font-size: 11px !important;
  pointer-events: auto !important;
  font-family: 'Inter', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto,
    Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  max-width: ${props => props.maxWidth || '200px'};
`

const Tooltip = props => {
  const { id, children, ...rest } = props

  return (
    <StyledTip id={id} delayHide={250} effect='solid' {...rest}>
      {children}
    </StyledTip>
  )
}

export default Tooltip
