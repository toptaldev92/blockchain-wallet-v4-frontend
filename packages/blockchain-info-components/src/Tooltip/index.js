import React from 'react'

import Tooltip from './template'
export { default as TooltipHost } from './TooltipHost'
export { default as TooltipIcon } from './TooltipIcon'

/**
 * In use, add as few tooltip containers as possible (they can be easily reused).
 * To give a component a tooltip, add the tag data-tip.
 * After it mounts, call TooltipRebuild to register it as a trigger.
 * Consult docs for further use https://github.com/wwayne/react-tooltip#react-tooltip
 *
 * TODO:
 * Add Proptypes
 * Default prop values in './template'
 */

export default class TooltipContainer extends React.PureComponent {
  render () {
    return <Tooltip {...this.props}>{this.props.children}</Tooltip>
  }
}
