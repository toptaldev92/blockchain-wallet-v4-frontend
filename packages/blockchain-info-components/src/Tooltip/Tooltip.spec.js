import React from 'react'
import Tooltip from './template.js'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

describe('Tooltip component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Tooltip colors={'primary'}><span>Default</span></Tooltip>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
