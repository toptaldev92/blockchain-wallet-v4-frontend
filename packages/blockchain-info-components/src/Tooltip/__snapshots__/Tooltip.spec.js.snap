// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component renders correctly 1`] = `
<div
  className="__react_component_tooltip place-top type-dark "
  data-id="tooltip"
  id="tip"
>
  <span>
    Tool Text
  </span>
</div>
`;
